modules/version-checker/version_checker_test.go

```diff
@@ -1,112 +1,219 @@
+		containError         bool
+		expectedErrorMessage string
+	}{
+		{
+			name:                 "Empty Params",
+			param:                CheckVersionParams{},
+			containError:         true,
+			expectedErrorMessage: "set WorkingDir in params",
+		},
+		{
+			name: "Missing VersionConstraint",
+			param: CheckVersionParams{
+				Binary:            Docker,
+				VersionConstraint: "",
+				WorkingDir:        ".",
+			},
+			containError:         true,
+			expectedErrorMessage: "set VersionConstraint in params",
+		},
+		{
+			name: "Invalid Version Constraint Format",
+			param: CheckVersionParams{
+				Binary:            Docker,
+				VersionConstraint: "abc",
+				WorkingDir:        ".",
+			},
+			containError:         true,
+			expectedErrorMessage: "invalid version constraint format found {abc}",
+		},
+		{
+			name: "Success",
+			param: CheckVersionParams{
+				Binary:            Docker,
+				VersionConstraint: ">1.2.3",
+				WorkingDir:        ".",
+			},
+			containError:         false,
+			expectedErrorMessage: "",
+		},
+	}
+
+	for _, tc := range tests {
+		err := validateParams(tc.param)
+		if tc.containError {
+			require.EqualError(t, err, tc.expectedErrorMessage, tc.name)
+		} else {
+			require.NoError(t, err, tc.name)
+		}
+	}
 }
 
 func TestExtractVersionFromShellCommandOutput(t *testing.T) {
 	t.Parallel()
 
-	// empty output
-	_, err := extractVersionFromShellCommandOutput("")
-	require.EqualError(t, err, "failed to find version using regex matcher")
-
-	// invalid version output from shell command
-	_, err = extractVersionFromShellCommandOutput("invalid output")
-	require.EqualError(t, err, "failed to find version using regex matcher")
+	tests := []struct {
+		name                 string
+		outputStr            string
+		expectedVersionStr   string
+		containError         bool
+		expectedErrorMessage string
+	}{
+		{
+			name:                 "Stand-alone version string",
+			outputStr:            "version is 1.2.3",
+			expectedVersionStr:   "1.2.3",
+			containError:         false,
+			expectedErrorMessage: "",
+		},
+		{
+			name:                 "version string with v prefix",
+			outputStr:            "version is v1.0.0",
+			expectedVersionStr:   "1.0.0",
+			containError:         false,
+			expectedErrorMessage: "",
+		},
+		{
+			name:                 "2 digit version string",
+			outputStr:            "version is v1.0",
+			expectedVersionStr:   "1.0",
+			containError:         false,
+			expectedErrorMessage: "",
+		},
+		{
+			name:                 "invalid output string",
+			outputStr:            "version is vabc",
+			expectedVersionStr:   "",
+			containError:         true,
+			expectedErrorMessage: "failed to find version using regex matcher",
+		},
+		{
+			name:                 "empty output string",
+			outputStr:            "",
+			expectedVersionStr:   "",
+			containError:         true,
+			expectedErrorMessage: "failed to find version using regex matcher",
+		},
+	}
+
+	for _, tc := range tests {
```

Comment:
Just a note that these tests are not running in parallel, unless I'm mistaken. However, they're quick and aren't doing anything like spinning up infrastructure, so running them serially is fine! 

If the `extractVersionFromShellCommandOutput` could become a long-running function it might be worth parallelizing. (Same with the `TestCheckVersionConstraint` and `TestCheckVersionEndToEnd` tests.)

For reference, to run them in parallel, you'd need to use `t.Run`, as in:
```go
for _, tc := range tests {
    t.Run(tc.name, func(t *testing.T) {
        tc := tc
        // run the test here...
    })
}
```